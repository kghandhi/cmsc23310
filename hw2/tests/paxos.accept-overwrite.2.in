# Same as paxos.accept-overwrite.1 but with Proposer 1
# and Acceptor 1 recovering at the end.
#
# Notice how Acceptor 1 had originally accepted value 42
# but its accepted value is legitimately overwritten because
# it agrees to participate in a higher-numbered ballot.

2 3 50
0 PROPOSE 1 42
8 FAIL PROPOSER 1
8 FAIL ACCEPTOR 1
10 PROPOSE 2 0
21 RECOVER PROPOSER 1
21 RECOVER ACCEPTOR 1
0 END
