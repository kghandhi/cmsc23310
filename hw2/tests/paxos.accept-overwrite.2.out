  0:    -> P1  PROPOSE v=42
  1: P1 -> A1  PREPARE n=1
  2: P1 -> A2  PREPARE n=1
  3: P1 -> A3  PREPARE n=1
  4: A1 -> P1  PROMISE n=1 (Prior: None)
  5: A2 -> P1  PROMISE n=1 (Prior: None)
  6: A3 -> P1  PROMISE n=1 (Prior: None)
  7: P1 -> A1  ACCEPT n=1 v=42
  8: ** P1 FAILS **
  8: ** A1 FAILS **
  9:
 10:    -> P2  PROPOSE v=0
 11: P2 -> A2  PREPARE n=2
 12: P2 -> A3  PREPARE n=2
 13: A2 -> P2  PROMISE n=2 (Prior: None)
 14: A3 -> P2  PROMISE n=2 (Prior: None)
 15: P2 -> A2  ACCEPT n=2 v=0
 16: P2 -> A3  ACCEPT n=2 v=0
 17: A2 -> P2  ACCEPTED n=2 v=0
 18: A3 -> P2  ACCEPTED n=2 v=0
 19:
 20:
 21: ** P1 RECOVERS **
 21: ** A1 RECOVERS **
 21: P1 -> A2  ACCEPT n=1 v=42
 22: P1 -> A3  ACCEPT n=1 v=42
 23: A1 -> P1  ACCEPTED n=1 v=42
 24: P2 -> A1  PREPARE n=2
 25: P2 -> A1  ACCEPT n=2 v=0
 26: A2 -> P1  REJECTED n=1
 27: A3 -> P1  REJECTED n=1
 28: A1 -> P2  PROMISE n=2 (Prior: n=1, v=42)
 29: A1 -> P2  ACCEPTED n=2 v=0
 30: P1 -> A1  PREPARE n=3
 31: P1 -> A2  PREPARE n=3
 32: P1 -> A3  PREPARE n=3
 33: A1 -> P1  PROMISE n=3 (Prior: n=2, v=0)
 34: A2 -> P1  PROMISE n=3 (Prior: n=2, v=0)
 35: A3 -> P1  PROMISE n=3 (Prior: n=2, v=0)
 36: P1 -> A1  ACCEPT n=3 v=0
 37: P1 -> A2  ACCEPT n=3 v=0
 38: P1 -> A3  ACCEPT n=3 v=0
 39: A1 -> P1  ACCEPTED n=3 v=0
 40: A2 -> P1  ACCEPTED n=3 v=0
 41: A3 -> P1  ACCEPTED n=3 v=0

P1 has reached consensus (proposed 42, accepted 0)
P2 has reached consensus (proposed 0, accepted 0)
